### PR TITLE
[18.01] Fix output discovery if dir doesn't exist

### PR DESCRIPTION
**Pull request by mvdbeek**

### CODE REVIEW

**Comment by martenson** [comment]
Aims to fix https://github.com/galaxyproject/galaxy/issues/5560

File: test/functional/tools/multi_output_recurse.xml [hunk 1]
```diff
@@ -23,6 +23,9 @@
     <data auto_format="true" name="report">
       <discover_datasets pattern="__designation__" directory="." visible="true" recurse="true" />
     </data>
+    <data auto_format="true" name="report_optional">
+      <discover_datasets pattern="__designation__" directory="i_do_not_exist" visible="true" recurse="true" optional="true" />
```
Comment (nsoranzo) [inline test/functional/tools/multi_output_recurse.xml]: From https://travis-ci.org/galaxyproject/galaxy/jobs/348335310 :
```
element discover_datasets: Schemas validity error : Element 'discover_datasets', attribute 'optional': The attribute 'optional' is not allowed.
```

**Comment by mvdbeek** [comment]
It does validate now, this is a different error here ~~(and a very strange one at that, it fails on something that isn't merged!).~~

This comes from https://github.com/galaxyproject/galaxy/pull/5589 ... weird that this hasn't errored out in that PR!

**Comment by mvdbeek** [comment]
OK, I see, it also didn't validate, I didn't notice the non-zero exit code when testing locally.

**Comment by jmchilton** [comment]
Thanks for the fix and the test case @mvdbeek, this looks great.

**Pull request merged**